Cesium3DTilesSelection/src/TilesetLoadIonAssetEndpoint.cpp

```diff
@@ -183,44 +243,25 @@ Tileset::LoadIonAssetEndpoint::Private::mainThreadHandleResponse(
     return tileset.getAsyncSystem().createResolvedFuture(
         std::make_optional(std::move(error)));
   }
-
-  auto pContext = std::make_unique<TileContext>();
-
-  pContext->pTileset = &tileset;
-  pContext->baseUrl = url;
-  pContext->requestHeaders.push_back(
-      std::make_pair("Authorization", "Bearer " + accessToken));
-  pContext->failedTileCallback = Private::onIonTileFailed;
-  return LoadTilesetDotJson::start(
-             tileset,
-             pContext->baseUrl,
-             pContext->requestHeaders,
-             std::move(pContext))
-      .thenImmediately([]() -> std::optional<TilesetLoadFailureDetails> {
-        return std::nullopt;
-      });
+  endpoint.type = type;
+  endpoint.url = url;
+  endpoint.accessToken = accessToken;
+  endpointCache[pRequest->url()] = endpoint;
+  return mainThreadLoadTilesetFromAssetEndpoint(tileset, endpoint);
 }
 
 namespace {
-
 /**
- * @brief Tries to update the context request headers with a new token.
+ * @brief Tries to obtain the `accessToken` from the JSON of the
+ * given response.
  *
- * This will try to obtain the `accessToken` from the JSON of the
- * given response, and set it as the `Bearer ...` value of the
- * `Authorization` header of the request headers of the given
- * context.
- *
- * @param pContext The context
  * @param pIonResponse The response
- * @return Whether the update succeeded
+ * @return The access token if successful
  */
-bool updateContextWithNewToken(
-    TileContext* pContext,
+std::optional<std::string> GetNewAccessToken(
```

Comment:
Use camelCase for function names.

```suggestion
std::optional<std::string> getNewAccessToken(
```